src/executor/evaluate/mod.rs

```diff
@@ -161,6 +165,45 @@ async fn evaluate_function<'a, T: 'static + Debug>(
 
             Ok(Evaluated::Value(value))
         }
+        name @ "LEFT" | name @ "RIGHT" => {
+            let convert = |string: String, number: usize| {
+                if name == "LEFT" {
+                    string.get(..number)
+                } else {
+                    let start_pos = if number > string.len() {
+                        0
+                    } else {
+                        string.len() - number
+                    };
+                    string.get(start_pos..)
+                }
+                .or(Some(&string))
+                .unwrap()
+                .to_string()
+            };
+
+            if args.len() != 2 {
+                return Err(EvaluateError::NumberOfFunctionParamsNotMatching {
+                    expected: 2,
+                    found: args.len(),
+                }
+                .into());
+            }
+
+            let string = match eval(&args[0]).await?.try_into()? {
+                Value::Str(string) | Value::OptStr(Some(string)) => Ok(string),
```

Comment:
https://github.com/gluesql/gluesql/pull/161

Yup :) You need to handle `Str` and `OptStr` differently.